WcaOnRails/app/javascript/show-schedule/index.js

```diff
@@ -0,0 +1,243 @@
+
+  // Setup events filter actions
+  $(".events-filter .event-all").click(onClickAllAction);
+  $(".events-filter span.cubing-icon").click(onClickEventIconAction);
+
+  // Setup rooms filter action
+  $(".toggle-room").click(onClickOnRoomAction);
+
+  // Hide popovers with round's information whenever we click somewhere
+  $('body').click(() => $(".has-popover").popover("hide"));
+}
+
+const getCalendarElemId = venueId => `#calendar-venue-${venueId}`;
+const getScheduleElemId = venueId => `#schedule-venue-${venueId}`;
+
+// Setup click event on "table" links (one for each venue)
+const onClickTableLinkAction = e => {
+  e.preventDefault();
+  let venueId = $(e.currentTarget).data("venue");
+  let idSelector = getScheduleElemId(venueId);
+  $(idSelector + " .schedule_calendar_container").hide();
+  $(idSelector + " .schedule-calendar-link").removeClass("active");
+  $(idSelector + " .schedule_table_container").show();
+  $(idSelector + " .schedule-table-link").addClass("active");
+}
+
+const onClickCalencarLinkAction = (popoverContentBuilder, locale, startDate, numberOfDays, e) => {
+  e.preventDefault();
+  let venueId = $(e.currentTarget).data("venue");
+  let idSelector = getScheduleElemId(venueId);
+  $(idSelector + " .schedule_table_container").hide();
+  $(idSelector + " .schedule-table-link").removeClass("active");
+  $(idSelector + " .schedule_calendar_container").show();
+  $(idSelector + " .schedule-calendar-link").addClass("active");
+  $calendar = $(getCalendarElemId(venueId));
+  if (!$calendar.hasClass("initialized")) {
+    $calendar.addClass("initialized");
+    let calendarParams = {
+      locale: locale,
+      venueId: venueId,
+      startDate: startDate,
+      numberOfDays: numberOfDays,
+      popoverContentBuilder: popoverContentBuilder,
+    };
+    initFullCalendar($calendar, calendarParams);
+  } else {
+    $calendar.fullCalendar("refetchEvents");
+  }
+}
+
+// Callback when the user click the "all" icon for events
+const onClickAllAction = e => {
+  // The "ALL" button
+  let $allButton = $(e.currentTarget);
+  // The whole filter with all events icons
+  let $filter = $allButton.parent();
+
+  if ($allButton.hasClass("selected")) {
+    // If "ALL" was active, we want to unselect all
+    $filter.find(".cubing-icon.selected").each(function(index, element) {
+      $element = $(element);
+      $element.removeClass("selected");
+      let eventId = $element.data("event");
+      // Table view action
+      $(".schedule-table .event-" + eventId).removeClass("event-selected");
+    });
+    $allButton.removeClass("selected");
+  } else {
+    $filter.find(".cubing-icon:not(.selected)").each(function(index, element) {
+      $element = $(element);
+      $element.addClass("selected");
+      let eventId = $element.data("event");
+      // Table view action
+      $(".schedule-table .event-" + eventId).addClass("event-selected");
+    });
+    $allButton.addClass("selected");
+  }
```

Comment:
Again it's definitely better since `toggleClass` can take an optional state parameter!
My initial concern was to be able to *set* the selected class on the icon and not *toggle* it, which is why I went with the code duplication.